components/encryption/src/io.rs

```diff
@@ -470,18 +496,20 @@ impl CrypterCore {
             buf[encrypted..target].copy_from_slice(&self.buffer[..crypter_count]);
             encrypted += crypter_count;
         }
+        self.offset += count as u64;
```

Comment:
question: Should self.offset be updated within while loop so that it can be recent 'encrypted' instead of 0, when it returns Err.

Reply:
The offset here is used to kept track of file writer offset, not the crypter internal offset. When file writer finishes writing a chunk of data, crypter must encrypt it atomically.